src/lib.rs

```diff
@@ -215,14 +220,44 @@ fn respond_to_request(stream: &mut TcpStream, build_dir_path: impl AsRef<Path>)
 }
 
 #[derive(Debug, StructOpt)]
-pub struct Watch {}
+pub struct Watch {
+    #[structopt(long = "watch", short = "w")]
+    watch_paths: Vec<PathBuf>,
+    #[structopt(long = "ignore", short = "i")]
+    exclude_paths: Vec<PathBuf>,
+}
 
 impl Watch {
-    pub fn execute(
-        &self,
-        watch_path: impl AsRef<Path>,
-        mut command: process::Command,
-    ) -> Result<()> {
+    pub fn new() -> Self {
+        Self {
+            exclude_paths: Vec::new(),
+            watch_paths: Vec::new(),
+        }
+    }
+
+    pub fn add_exclude_path(&mut self, path: impl AsRef<Path>) {
+        self.exclude_paths.push(path.as_ref().to_path_buf())
+    }
```

Comment:
If the path is provided by the end user, it's probably relative to the current directory.

On the other hand, if the path is provided by the xtask developer and is relative, then it is relative to the workspace directory.

Therefore you should actually have 2 exclude_paths fields:
1. one that is for structopt (for the end user)
2. one that is invisible to structopt (`#[structopt(skip)]`) for the xtask developer

The build directory for example is relative to the workspace directory.